snuba/consumers/consumer.py

```diff
@@ -355,6 +360,66 @@ def build_writer() -> ProcessedMessageBatchWriter:
     return build_writer
 
 
+class DeadLetterStep(
+    ProcessingStep[Tuple[StorageKey, Union[None, BytesInsertBatch, ReplacementBatch]]]
+):
+    def __init__(self, producer: AbstractProducer[KafkaPayload], topic: Topic):
+        self.__producer = producer
+        self.__topic = topic
+        self.__futures: Deque[Future[Message[KafkaPayload]]] = deque()
+        self.__closed = False
+
+    def __format_payload(
+        self,
+        message: Message[
+            Tuple[StorageKey, Union[None, BytesInsertBatch, ReplacementBatch]]
+        ],
+    ) -> Optional[KafkaPayload]:
+        kafka_payload = None
+        storage_key, payload = message.payload
+        if isinstance(payload, BytesInsertBatch):
+            rows = [rapidjson.loads(row) for row in payload.rows]
```

Comment:
You should not need to parse the JSON.
Just take binary data from  `payload.rows` and produce them as they are. No parsing.